src/App.tsx

```diff
@@ -15,36 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  BY_ALPHABET = 'alphabet',
+  BY_LENGTH = 'length',
+  DEFAULT = '',
+}
+
+interface FilterParams {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: FilterParams,
+) {
+  const copiedGoods = [...goods];
+
+  if (sortField) {
+    copiedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    return copiedGoods.reverse();
+  }
+
+  return copiedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
+
+  const reset = () => {
```

Comment:
use best practices for handlers
```suggestion
  const handleReset = () => {
```